server/src/main/scala/services/graphstages/ApproximateScheduleMatch.scala

```diff
@@ -0,0 +1,63 @@
+package services.graphstages
+
+import drt.shared.api.Arrival
+import drt.shared.{MilliTimes, PortCode, UniqueArrival}
+import org.slf4j.{Logger, LoggerFactory}
+import services.arrivals.LiveArrivalsUtil
+
+
+object ApproximateScheduleMatch {
+  val log: Logger = LoggerFactory.getLogger(getClass)
+
+  def findInScheduledWindow(uniqueArrival: UniqueArrival,
+                            origin: PortCode,
+                            arrivalsToSearch: Map[UniqueArrival, Arrival],
+                            windowMillis: Int): Either[Arrival, List[Arrival]] = {
+    val approxMatches = arrivalsToSearch
+      .filterKeys(_.equalWithinScheduledWindow(uniqueArrival, windowMillis))
+      .filter {
+        case (_, potentialArrival) => potentialArrival.Origin == origin
```

Comment:
Here ,  Would it be nice to log if match didn't happen due to Origin so we understand if origin is important to considered or understand same flight number could have different origin within an hour ?

Reply:
I'm not sure we have a case where we'd ever want to consider matching arrivals from different origins as they'd be physically different flights

Reply:
it would interesting to know flight that same number , terminal and schedule approximation within hour to have different origin .  I am thinking in terms of knowing more information about such flights and fact that flight numbers are not universal when terminal and schedule time are almost similar.